contracts/smart-contract-wallet/SmartAccount.sol

```diff
@@ -641,25 +650,38 @@ contract SmartAccount is
     }
 
     /**
-     * execute a transaction (called directly from owner, or by entryPoint)
+     * Execute a transaction (called directly from owner, or by entryPoint)
+     * @dev name is optimized for this method to be cheaper to be called
      */
-    function executeCall(
+    function executeCall_s1m(
```

Comment:
do add the test case for calling this method. in aa transactions calldata is prepared for executeCall and executeBatchCall, replace these instead.

Reply:
can we also manage to get some gas usage numbers for the changes?

Reply:
+

executeCall · 39107 · 169358 · **65077**
executeCall_s1m · 39085 · 169336 · **65055**
**22 gas less on average**

executeBatchCall · 52026 · 75260 · **63739**
executeBatchCall_4by · 51972 · 75206 · **63685**
**54 gas less on average**